packages/resolve-query/src/index.js

```diff
@@ -3,32 +3,20 @@ import 'regenerator-runtime/runtime';
 function getExecutor({ eventStore, projection }) {
     const eventTypes = Object.keys(projection.eventHandlers);
     let state = projection.initialState || {};
-    let error = null;
 
-    const eventStream = eventStore.getStreamByEventTypes(eventTypes);
+    let result = null;
 
-    eventStream.on('readable', () => {
-        let event;
-        // eslint-disable-next-line no-cond-assign
-        while (null !== (event = eventStream.read())) {
-            const handler = projection.eventHandlers[event.type];
-            if (!handler) continue;
+    return async () => {
+        result =
+            result ||
+            eventStore.subscribeByEventType(eventTypes, (event) => {
+                const handler = projection.eventHandlers[event.type];
+                if (!handler) return;
 
-            try {
                 state = handler(state, event);
-            } catch (err) {
-                error = err;
-            }
-        }
-    });
-
-    eventStream.on('error', err => (error = err));
+            });
 
-    const storageReadyPromise = new Promise(resolve => eventStream.on('storageDone', resolve));
-
-    return async () => {
-        if (error) throw error;
-        await storageReadyPromise;
+        await result;
         return state;
```

Comment:
In particular, if using resolve-bus-memory, call stack will be crashed and contain this line https://github.com/reimagined/resolve/blob/0bc136c5356d29f12819c6642e4f3d1d971f3282/packages/resolve-bus/src/index.js#L6
If using resolve-storage-mongo, it will be crashed at mongodb-core/lib/connection/connection.js line 10291 and below